R/surv-concordance.R

```diff
@@ -0,0 +1,113 @@
+#' Concordance index for right-censored data
+#'
+#' Compute the Concordance index for right-censored data
+#'
+#' @family static survival metrics
+#' @templateVar fn concordance_survival
+#' @template return
+#' @details
+#'
+#' The concordance index is defined as the proportion of all comparable pairs in
+#' which the predictions and outcomes are concordant.
+#'
+#' Two observations are comparable if:
+#'
+#' 1. both of the observations experienced an event (at different times), or
+#' 2. the observation with the shorter observed survival time experienced an
+#'    event, in which case the event-free subject “outlived” the other.
+#'
+#' A pair is not comparable if they experienced events at the same time.
+#'
+#' Concordance intuitively means that two samples were ordered correctly by the
+#' model. More specifically, two samples are concordant, if the one with a
+#' higher estimated risk score has a shorter actual survival time.
+#'
+#' Larger values of the score are associated with better model performance.
+#'
+#' @inheritParams pr_auc
+#'
+#' @param data A `data.frame` containing the columns specified by `truth` and
+#'   `estimate`.
+#'
+#' @param truth The column identifier for the true class survival result (that
+#'   is created using [survival::Surv()].). This should be an unquoted column
+#'   name although this argument is passed by expression and supports
+#'   [quasiquotation][rlang::quasiquotation] (you can unquote column names). For
+#'   `_vec()` functions, an [survival::Surv()] object.
+#'
+#' @param estimate The column identifier for the survival probabilities. This
+#'   needs to be a numeric vector.This should be an unquoted column name
+#'   although this argument is passed by expression and supports
+#'   [quasiquotation][rlang::quasiquotation] (you can unquote column names). For
+#'   `_vec()` functions, a numeric vector.
```

Comment:
Could these be inherited from some other documentation? The Brier score, maybe, at least for `truth` and `estimate`?

Reply:
done!